zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceWithTerminationInstructionTest.java

```diff
@@ -271,4 +271,60 @@ record ->
             Tuple.tuple(
                 ProcessInstanceIntent.ELEMENT_ACTIVATING, boundaryEventNestedSubProcessEnd));
   }
+
+  @Test
+  public void shouldTerminateOnlyOnceForMultiInstanceElements() {
+    // given
+    final String processId = "process";
+    final String multiInstanceTaskId = "multiInstanceTask";
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(processId)
+                .startEvent()
+                .scriptTask()
+                .id(multiInstanceTaskId)
+                .zeebeExpression("i + 2")
+                .zeebeResultVariable("j")
+                .multiInstance(
+                    b ->
+                        b.zeebeInputCollectionExpression("[1,2,3]")
+                            .zeebeInputElement("i")
+                            .zeebeOutputCollection("out")
+                            .zeebeOutputElementExpression("j + 2"))
```

Comment:
🔧 This is not necessary for the test case. Let's remove it to focus the test case.